IdentityCore/src/requests/sdk/MSIDBrokerResponseHandler.m

```diff
@@ -122,7 +122,21 @@ - (MSIDTokenResult *)handleBrokerResponseWithURL:(NSURL *)response error:(NSErro
         if (error) *error = brokerError;
         return nil;
     }
-
+    
+    NSString *applicationToken = brokerResponse.applicationToken;
+    
+    if (![NSString msidIsStringNilOrBlank:applicationToken])
+    {
+        NSError *appTokenError = nil;
+        BOOL saveAppToken = [brokerKeyProvider saveApplicationToken:applicationToken forClientId:brokerResponse.clientId error:&appTokenError];
+        
+        if (!saveAppToken)
+        {
+            if (error) *error = appTokenError;
```

Comment:
I don't think we should fail the whole request if we fail to save token. We should still log it, but this token is the best effort case and user experience improvement. We can still proceed even if we fail to save it.

Reply:
Addressed.